src/bm3dornl/bm3d.py

```diff
@@ -739,26 +739,44 @@ def bm3d_ring_artifact_removal_ms(
             filter_kwargs=filter_kwargs,
         )
 
-    # step 1: create a list of binned sinograms
-    binned_sinos = horizontal_binning(sinogram, k=k)
-    # reverse the list
-    binned_sinos = binned_sinos[::-1]
-
-    # step 2: estimate the noise level from the coarsest sinogram, then working back to the original sinogram
-    noise_estimate = None
-    for i in range(len(binned_sinos)):
-        logging.info(f"Processing binned sinogram {i+1} of {len(binned_sinos)}")
-        sino = binned_sinos[i]
-        sino_star = (
-            sino if i == 0 else sino - horizontal_debinning(noise_estimate, sino)
+    denoised_sino = None
+    # Make a copy of an original sinogram
+    sino_orig = horizontal_binning(sino_star, 1, dim=0)
+    binned_sinos_orig = [np.copy(sino_orig)]
+
+    # Contains upscaled denoised sinograms
+    binned_sinos = [np.zeros(0)]
+
+    # Bin horizontally
+    for i in range(0, k):
+        binned_sinos_orig.append(horizontal_binning(binned_sinos_orig[-1], 2, dim=1))
+        binned_sinos.append(np.zeros(0))
+
+    binned_sinos[-1] = binned_sinos_orig[-1]
+
+    for i in range(k, -1, -1):
+        logging.info(f"Processing binned sinogram {i + 1} of {k}")
+        # Denoise binned sinogram
+        denoised_sino = bm3d_ring_artifact_removal(
+            binned_sinos[i],
+            mode=mode,
+            block_matching_kwargs=block_matching_kwargs,
+            filter_kwargs=filter_kwargs,
         )
-
-        if i < len(binned_sinos) - 1:
-            noise_estimate = sino - bm3d_ring_artifact_removal(
-                sino_star,
-                mode=mode,
-                block_matching_kwargs=block_matching_kwargs,
-                filter_kwargs=filter_kwargs,
+        # For iterations except the last, create the next noisy image with a finer scale residual
+        if i > 0:
+            debinned_sino = horizontal_debinning(
+                denoised_sino - binned_sinos_orig[i],
+                binned_sinos_orig[i - 1].shape[1],
+                2,
+                30,
```

Comment:
hard-coded magic numbers should be avoided, or at least provide some comments on the reasoning.

Reply:
Added function parameters

Reply:
can you explain a bit on the decision on selecting `2` and `30` here? This is to help the future developer so that they don't have to guess why these number are chosen.